assets/src/models/ladderVehiclePosition.ts

```diff
@@ -0,0 +1,156 @@
+import { LadderDirection } from "../components/ladder"
+import {
+  centerOfVehicleGroupY,
+  heightOfVehicleGroup,
+} from "../components/ladderVehicle"
+import { Vehicle } from "../skate"
+
+export enum VehicleDirection {
+  Up,
+  Down,
+}
+
+export interface LadderVehiclePosition {
+  vehicle: Vehicle
+  x: number
+  y: number
+  vehicleDirection: VehicleDirection
+  lane?: number
+}
+
+type Range = [number, number]
+
+export const vehicleDirectionOnLadder = (
+  vehicle: Vehicle,
+  ladderDirection: LadderDirection
+): VehicleDirection =>
+  (vehicle.direction_id === 1) ===
+  (ladderDirection === LadderDirection.ZeroToOne)
+    ? VehicleDirection.Down
+    : VehicleDirection.Up
+
+export const putIntoLanes = (
+  ladderVehiclePositions: LadderVehiclePosition[]
+): LadderVehiclePosition[] =>
+  ladderVehiclePositions
+    .slice()
+    .sort(byDirectionAndY)
+    .reduce(
+      (
+        previousLadderVehiclePositions: LadderVehiclePosition[],
+        ladderVehiclePosition: LadderVehiclePosition
+      ): LadderVehiclePosition[] => {
+        const overlappers = overlappingPreviousVehicles(
+          previousLadderVehiclePositions,
+          ladderVehiclePosition
+        )
+        const occupiedLanes: number[] = overlappers.reduce(
+          (lanes: number[], overlapper: LadderVehiclePosition) =>
+            overlapper.lane !== undefined ? [...lanes, overlapper.lane] : lanes,
+          []
+        )
+        const lane = firstOpenLane(occupiedLanes)
+
+        return [
+          {
+            ...ladderVehiclePosition,
+            lane,
+          },
+          ...previousLadderVehiclePositions,
+        ]
+      },
+      []
+    )
+
+export const xInLaneWithOffset = (ladderVehicleHorizontalOffset: number) => ({
+  x,
+  vehicleDirection,
+  lane,
+}: LadderVehiclePosition): number => {
+  const laneOffset = (lane || 0) * ladderVehicleHorizontalOffset
+  return vehicleDirection === VehicleDirection.Up
+    ? x + laneOffset
+    : x - laneOffset
+}
+
+export const byDirectionAndY = (
+  a: LadderVehiclePosition,
+  b: LadderVehiclePosition
+): number => {
+  if (a.vehicleDirection === VehicleDirection.Down) {
+    if (b.vehicleDirection === VehicleDirection.Up) {
+      return -1
+    } else {
+      if (a.y > b.y) {
+        return -1
```

Comment:
This comparison is a little hard to follow. It could use a quick human description "Down is less than Up. If they're the same direction the vehicle behind is less than the vehicle in front" or something similar (idk if I got the directions right)

What impact does the order of the sorting have? If it's important, that should probably get noted too.